client/src/api/utils/AuthUtils.ts

```diff
@@ -0,0 +1,37 @@
+import { AccountInfo, IPublicClientApplication } from "@azure/msal-browser";
+import axios from "axios";
+
+export class AuthUtil {
+    private instance: IPublicClientApplication;
+    private account: AccountInfo;
+
+    constructor(instance: IPublicClientApplication, account: AccountInfo) {
+      this.instance = instance;
+      this.account = account;
+    }
+
+    async getAccessToken() {
+      const api_scope = 'api://' + import.meta.env.IQENGINE_APP_ID + '/api';
+      if (!this.account) return null;
+      try {
+        const response = await this.instance.acquireTokenSilent({
+          account: this.account,
+          scopes: [api_scope],
+        });
+        return response.accessToken;
+      } catch (error) {
+        return null;
+      }
+    }
+
+    async requestWithAuthIfRequired(config) {
+      const token = await this.getAccessToken();
```

Comment:
We should always require the cancelation token in the request otherwise we will endup with many requests in the client eve after the customer has been disconnected.